ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,228 @@
+	defer ctrl.Finish()
+
+	ackSent := make(chan *ecsacs.AckRequest)
+
+	testResponseSender := func(response interface{}) error {
+		resp := response.(*ecsacs.AckRequest)
+		ackSent <- resp
+		return nil
+	}
+
+	testTaskManifestResponder := setupTestTaskManifestResponder(t, mock_metrics.NewMockEntryFactory(ctrl), testResponseSender)
+
+	// After we get the latest sequence number, we will expect to set the latest
+	// message ID
+	mockmmIDA := testTaskManifestResponder.messageIDAccessor.(*mock_session.MockManifestMessageIDAccessor)
+	mockmmIDA.EXPECT().SetMessageID(gomock.Any()).Times(1)
+
+	// Expect to get the latest sequence number for stale check, then set it as we expect
+	// to pass
+	mockSNA := testTaskManifestResponder.snAccessor.(*mock_session.MockSequenceNumberAccessor)
+	mockSNA.EXPECT().GetLatestSequenceNumber().Times(1)
+	mockSNA.EXPECT().SetLatestSequenceNumber(gomock.Any()).Times(1)
+
+	// Test Metric publication when our responder is initialized with a mock entry factory
+	mockMetrics := testTaskManifestResponder.metricsFactory.(*mock_metrics.MockEntryFactory)
+	mockEntry := mock_metrics.NewMockEntry(ctrl)
+	mockMetrics.EXPECT().New(metrics.TaskManifestHandlingDuration).Return(mockEntry).Times(1)
+	mockEntry.EXPECT().Done(nil).Times(1)
+
+	// Create test task manifest.
+	testManifest := setupTestManifestMessage()
+
+	mockComparer := testTaskManifestResponder.taskComparer.(*mock_session.MockTaskComparer)
+	mockComparer.EXPECT().CompareRunningTasksOnInstanceWithManifest(gomock.Any()).Times(1)
+
+	// Handle the task manifest message update
+	handleManifestMessage := testTaskManifestResponder.HandlerFunc().(func(*ecsacs.TaskManifestMessage))
+	go handleManifestMessage(testManifest)
+
+	// Wait for the ack to send through our channel so we can receive it
+	manifestMessageAckSent := <-ackSent
+	require.Equal(t, aws.StringValue(manifestMessageAckSent.MessageId), aws.StringValue(testManifest.MessageId))
+
+}
+
+// TestTaskManifestSequenceNumbers will make sure the manifest can update, retrieve and catch stale
+// Manifest sequence numbers appropriately.
+func TestTaskManifestSequenceNumbers(t *testing.T) {
+
+	// Create a waitgroup to make sure the sequence number has updated
+	// before we try to access it
+	var ackWg sync.WaitGroup
+	ackWg.Add(1)
+
+	testResponseSender := func(response interface{}) error {
+		defer ackWg.Done()
+		return nil
+	}
+
+	testTaskManifestResponder := setupTestTaskManifestResponder(t, metrics.NewNopEntryFactory(), testResponseSender)
+
+	mockmmIDA := testTaskManifestResponder.messageIDAccessor.(*mock_session.MockManifestMessageIDAccessor)
+	mockmmIDA.EXPECT().SetMessageID(gomock.Any()).Times(1)
+
+	// Expect GetLatestSequenceNumber once in handleManifestMessage, and once
+	// when we validate it
+	mockSNA := testTaskManifestResponder.snAccessor.(*mock_session.MockSequenceNumberAccessor)
+	mockSNA.EXPECT().GetLatestSequenceNumber().Times(2)
+	mockSNA.EXPECT().SetLatestSequenceNumber(aws.Int64(20)).Times(1)
```

Comment:
nit: Can we create a const or variable with an appropriate name for `aws.Int64(20)` to know what it is in this context?

Reply:
ack